hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/HederaState.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state;
+
+import com.hedera.node.app.spi.state.ReadableStates;
+import com.hedera.node.app.spi.state.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * The full state used by Hedera. The current implementation is based on a merkle tree, and the data
+ * structures provided by the hashgraph platform. But most of our code doesn't need to know that
+ * detail, and are happy with just the API provided by this interface.
+ */
+public interface HederaState {
+    /**
+     * Creates a {@link ReadableStates} for the given named service. If such a service doesn't
+     * exist, an empty {@link ReadableStates} is returned.
+     *
+     * @param serviceName The name of the service.
+     * @return A collection of {@link com.hedera.node.app.spi.state.ReadableState} instance
+     *     belonging to the service.
+     */
+    @NonNull
+    ReadableStates createReadableStates(@NonNull String serviceName);
```

Comment:
Why did you pick `String` to identify a service? If we keep `String`, the names should be defined somewhere.

Reply:
I think we talked about this offline, but to record for posterity:

It really comes down to a balance that I don't think we've found yet. Ideally all services (or maybe all services except the crypto service) would be plugins, and the app framework wouldn't know anything about them. For that to work, we need some common type that would apply to all of them, and I guess String is as good as any.

In this current implementation and design, we don't achieve this goal, and in fact we could use an enum or something. But I guess String is OK.